command/agent/job_endpoint.go

```diff
@@ -310,3 +347,86 @@ func (s *HTTPServer) jobDispatchRequest(resp http.ResponseWriter, req *http.Requ
 	setIndex(resp, out.Index)
 	return out, nil
 }
+
+func (s *HTTPServer) apiJobToStructJob(job *api.Job) *structs.Job {
+	if job.Priority == nil {
```

Comment:
Why is this not api.Canonicalize() and then just setting each field. It is probably also worth adding apiJobToStructJob which calls apiTaskGroupsToStructTaskGroups and so on.